### PR TITLE
chore(ci): every hour run tests and report failure to #alerts

### PR DESCRIPTION
**Pull request by dpc**
I think we got flakes under control, but we must remain vigilant.

This workflow will give us 24 run a day, to make sure any new flakes us quickly spotted.

### CODE REVIEW

**Comment by dpc** [comment]
@elsirion It seems like a good idea, and the only potential downside is an extra buildjet cost.

File: .github/workflows/hourly-run.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        with:
+          name: fedimint
+          authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+        continue-on-error: true
+
+      - name: Tests (5 times more)
+        run: |
+          set -euo pipefail
+          # first one might do nothing
+          nix build -L .#wasm32-unknown.ci.ciTestAll5Times --keep-failed
```
Comment (bradleystachurski) [inline .github/workflows/hourly-run.yml:57]: Interesting, ciTestAll5Times will sometimes not run? Same for wasmTest? Or is this actually a ciTestAll10Times?

File: .github/workflows/hourly-run.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+  test:
+    if: github.repository == 'fedimint/fedimint'
+    strategy:
+      matrix:
+        host:
+          - linux
+        include:
+          - host: linux
+            runs-on: buildjet-8vcpu-ubuntu-2004
```
Comment (bradleystachurski) [inline .github/workflows/hourly-run.yml:29]: [nit] Should we consider bumping our buildjet runners from 2004 to 2204?

**Review by bradleystachurski — DISMISSED** [review]
CI is in a healthy state and high signal, so I think this is a great idea 👍 

Back-of-the-envelope costs:
- ~50min for ciTestAll5times and wasmTest ([recent run](https://github.com/fedimint/fedimint/actions/runs/8351416363/job/22859711551))
- $0.016/min `buildjet-8vcpu-ubuntu-2204` ([pricing](https://buildjet.com/for-github-actions/docs/about/pricing))
- 24 runs/day
- ~$19.20/day in buildjet costs

**Comment by elsirion** [comment]
Can we maybe run a bit less often, every 2-3h or so? This sounds like a great reason to switch to self-hosted build infra that is just there 24/7, then we could just run this whenever the load is low.

File: .github/workflows/hourly-run.yml (hunk 1 above)
Comment (dpc) [reply .github/workflows/hourly-run.yml:57]: It's already in the Nix cache, so Nix will just not run it. :shrug: 

We just ensure it's there, then we can run with `--rebuild` which will run it.

File: .github/workflows/hourly-run.yml (hunk 2 above)
Comment (dpc) [reply .github/workflows/hourly-run.yml:29]: Ubuntu 20.04 :D

**Comment by dpc** [comment]
Switched it to run on free ubuntu boxes.

**Review by bradleystachurski — APPROVED** [review]

**Review by maan2003 — APPROVED** [review]
lets see how it goes, we can always revert

**Comment by dpc** [comment]
> lets see how it goes, we can always revert

That's the right attitude! :)

**Pull request merged**